doc/ref_arch/kubernetes/chapters/chapter04.md

```diff
@@ -79,8 +98,16 @@ In order for the Kubernetes components to be conformant with the Reference Archi
 |`ra2.k8s.004`|Highly available worker nodes|An implementation must consist of at least one worker node per availability zone or fault domain to ensure the high availability and resilience of workloads managed by Kubernetes|[req.gen.rsl.01 req.gen.avl.01 req.kcm.gen.02](./chapter02.md#23-kubernetes-architecture-requirements)|
 |`ra2.k8s.005`|Kubernetes API Version|In alignment with the [Kubernetes version support policy](https://kubernetes.io/docs/setup/release/version-skew-policy/#supported-versions), an implementation must use one of three latest minor versions (`n-2`). e.g. if the latest version is 1.17 then the RI must use either 1.17, 1.16 or 1.15.|TBC|
-|`ra2.k8s.006`||||
-|`ra2.k8s.007`||||
+|`ra2.k8s.006`|NUMA Support|The `TopologyManager` and `CPUManager` feature gates must be enabled and configured on the kubelet (note, TopologyManager is enabled by default in Kubernetes v1.18 and later, with CPUManager enabled by default in Kubernetes v1.10 and later). `--feature-gates="...,TopologyManager=true,CPUManager=true" --topology-manager-policy=single-numa-node --cpu-manager-policy=static`|[e.cap.007](./chapter02.md#221-cloud-infrastructure-software-profile-capabilities) [infra.com.cfg.002](./chapter02.md#223-cloud-infrastructure-software-profile-requirements) [infra.hw.cpu.cfg.004](./chapter02.md#224-cloud-infrastructure-hardware-profile-requirements)|
+|`ra2.k8s.007`|DevicePlugins Feature Gate|In order to support the various acceleration devices required, the DevicePlugins feature gate must be enabled (note, this is enabled by default in Kubernetes v1.10 or later). `--feature-gates="...,DevicePlugins=true,..."`|Various, e.g. [e.cap.013, e.cap.014](./chapters/chapter02.md#221-cloud-infrastructure-software-profile-capabilities)|
+|`ra2.k8s.008`|System Resource Reservations|To avoid resource starvation issues on nodes, reserve compute resources for system daemons and Kubernetes system daemons such as kubelet, container runtime, etc. (requires Kubernetes version 1.17 or later). Use the following kubelet flags: `--reserved-cpus=[0-3]`|NB: There doesn't appear to be a requirement that this is for?  Do we need to be specific about the values used?  The guidelines [here](https://kubernetes.io/docs/tasks/administer-cluster/reserve-compute-resources/#general-guidelines) suggest a more nuanced approach is recommended?|
+|`ra2.k8s.009`|CPU Pinning|In order to support CPU Pinning, the kubelet must be started with the `--cpu-manager-policy=static` option. (Note, only containers in `Guaranteed` pods - where CPU resource `requests` and `limits` are identical - and configured with positive-integer CPU `requests` will take advantage of this. All other Pods will run on CPUs in the remaining shared pool.)|[infra.com.cfg.003](./chapter02.md#223-cloud-infrastructure-software-profile-requirements)|
+|`ra2.k8s.010`||||
```

Comment:
reserving CPU is nice, how about system reserving memory too?

Reply:
Don't think it's possible in Kubernetes, but I agree - it's as important.

Reply:
--system-reserved Kubelet flag :)

Reply:
We need to configure [hugepages](https://kubernetes.io/docs/tasks/manage-hugepages/scheduling-hugepages/).
OVS-DPDK: need to pre-allocate from hugepages on specific sockets for running OvS-DPDK processes; I understand, we can specify up to 8 1G hugepages using "dpdk-socket-mem=".

Reply:
@pgoyal01 - `ra2.ch.001` should cover configuration within the host OS.  Let me know if that's not quite right.  The requirements I pulled through into chapter02.md didn't include a number of each hugepage size, but if there are other requirements that require a particular number of them then we need to include that and reference those reqs from chapter02.md as well.

> |`ra2.ch.001`|Huge Pages|It must be possible to enable Huge Pages (2048KiB and 1048576KiB) within the Container Host OS, exposing schedulable resources `hugepages-2Mi` and `hugepages-1Gi`.|

Reply:
@tomkivlin I was just replying to the previous comments about memory reservation.  Ra2.ch.001 is correct. Thanks